test/test_model_relation_data_class.py

```diff
@@ -122,8 +122,8 @@ def json_nested_hook(dct: dict[str, Any]) -> Nested | dict[str, Any]:
 
 
 class MyCharm(BaseTestCharm):
-    encoder = nested_encode
-    decoder = nested_decode
+    encoder = staticmethod(nested_encode)
```

Comment:
Thanks for doing all of these. This finally gets rid of all (but one of) the deprecation warnings when running the tests, which have annoyed me for ages but not enough to actually fix them!